realm/realm-library/src/main/java/io/realm/RealmModel.java

```diff
@@ -0,0 +1,28 @@
+package io.realm;
+
+import io.realm.annotations.RealmClass;
+
+
+/**
+ * Interface for marking classes as RealmObjects, it can be used instead of extending {@link RealmObject}.
+ *
+ * All helper methods available to classes that extend RealmObject are instead available as static methods:
+ *
+ * <pre>
+ * {@code
+ *   Person p = realm.createObject(Person.class);
+ *
+ *   // With the RealmModel interface
+ *   RealmObject.isValid(p);
```

Comment:
Not sure how I feel about this yet. I understand that this is the penalty for not extending a base class. Still works though ... the developer using the API needs to determine the tradeoffs.